README.md

```diff
@@ -1 +1,375 @@
+
+``` mermaid
+flowchart
+    a --> b
+    a --> c
+    b --> d
+    c -.->|ref| d
+    d --> e
+```
+
+## 3.3 Tabling
+
+Included here for completeness, a tabling strategy SHOULD be used when no inlining is desired. This is often the most efficient strategy for storage and retrieval of cross-linked data.
+
+``` mermaid
+flowchart LR
+    car --> a
+    a -.->|ref| b
+
+    car --> b
+
+    car --> c
+    b ~~~ c
+    a -.->|ref| c
+    
+    car --> d
+    b -.->|ref| d
+    c -.->|ref| d
+
+    car --> e
+    d -.->|ref| e
+```
+
+# 4 IPLD Canonicalization
+
+Inline links MUST NOT be used in calculation of CIDs. The only time that inline links interact with CIDs MUST be when inheriting the CID configuration from the parent context.
+
+To calculate the CID of a DAG that contains inline links, first walk the graph and replace all inline links with regular CID links (i.e. of the format `{"/": <cid>}`). This MUST be performed recursively when encountering an inline link. Once converted to normal IPLD, proceed as normal.
+
+## 4.1 Example
+
+``` js
+// Implicit inline inside DAG-CBOR with SHA2-256
+{
+  "name": "Alonzo Church",
+  "birthday": {
+    "/": {
+      "cid": null,
+      "dag": {
+        "day": 14,
+        "month": 6
+      }
+    }
+  }
+}
+
+// Converted to explicit CID
+{
+  "name": "Alonzo Church",
+  "birthday": {
+    "/": {
+      "cid": "bafyreif7dowvi5nuzzijawl22vpqsughufapj455diyflrk7htswzbjid4",
+      "dag": {
+        "day": 14,
+        "month": 6
+      }
+    }
+  }
+}
+
+// Canonical
+{
+  "birthday": {"/": "bafyreif7dowvi5nuzzijawl22vpqsughufapj455diyflrk7htswzbjid4"},
+  "name": "Alonzo Church"
+}
+```
+
+# 5 FAQ
+
+## 5.1 Why Shallow Nesting?
+
+The two options explored in this design were nesting the link under a `"/"` key, or adding another key as a sibling at the same level. For these reasons given below, it was decided that the nested strategy is the least likely to be misinterpreted or misimplemented.
+
+### 5.1.1 Sibling DAG
+
+``` json
+{ 
+  "/": "bafyreif7dowvi5nuzzijawl22vpqsughufapj455diyflrk7htswzbjid4",
+  ".": {
+    "day": 14,
+    "month": 6
+  }
+}
+
+{ 
+  "/": null,
+  ".": {
+    "day": 14,
+    "month": 6
+  }
+}
+```
+
+The advantageous features of the sibling strategy include:
+
+- The link (when present) looks exactly like a normal CID
+- It saves a few characters in JSON
+
+### 5.1.2 Shallowly Nested
+
+This is the one described in the specification.
+
+``` json
+{ 
+  "/": {
+    "cid": "bafyreif7dowvi5nuzzijawl22vpqsughufapj455diyflrk7htswzbjid4",
+    "dag": {
+      "day": 14,
+      "month": 6
+    }
+  }
+}
+
+{ 
+  "/": {
+    "cid": "null",
+    "dag": {
+      "day": 14,
+      "month": 6
+    }
+  }
+}
+```
+
+The advantages of the nesting strategy include:
+
+- Distinguishes clearly the case where a `null` CID would otherwise be a parse error
```

Comment:
The argumentis something like there being more error cases:

- The parser can accdientally skip the `data` field and only parse out the `/` field
- `{"/": null}` with no DAG should fail, so you have to model inlines/embeds as something further from their presentation in various langauge libraries

`{"/": ..., ".": ...}` was actually my first version, because I didn't want to have to follow as many deeply nested links and it "feels" like an extension to normal CIDs. That punning ended up feeling less clear and more prone to error. But, I figured that presentation would be the biggest part of this conversation, so let's chat about it! DO you have a preference?

Reply:
> The parser can accdientally skip the data field and only parse out the / field

This is arguably a feature IMO, I could pass data into e.g. DAG-JSON codec and it will correctly resolve the block ignoring the data field.

Although it is true that if `data` and `/` mismatch block codec my succeed while, dag codec should fail.

That said, I do think that understand the rational and I think it makes sense.